### ISSUE TITLE
Recording does not work

### ISSUE THREAD

**Issue by nucombo** [jfversluis/Plugin.Maui.Audio#145]
I am trying your sample with a brand new MAUI on .NET 9 nothing special in it just play simple:

```cs
using Plugin.Maui.Audio;

namespace MauiApp1
{
    public partial class MainPage : ContentPage
    {
        private readonly IAudioManager _audioManager;
        private readonly IAudioRecorder _audioRecorder;

        public MainPage(IAudioManager audioManager)
        {
            InitializeComponent();
            _audioManager = audioManager;
            _audioRecorder = _audioManager.CreateRecorder();
        }

        private async void StopRecording_Clicked(object sender, EventArgs e)
        {
            StopRecording.IsEnabled = false;
            var audioSource = await _audioRecorder.StopAsync();
            var player = _audioManager.CreateAsyncPlayer(((FileAudioSource)audioSource).GetAudioStream());
            await player.PlayAsync(CancellationToken.None);
            StartRecording.IsEnabled = true;
            StopRecording.IsEnabled = true;
        }

        private async void StartRecording_Clicked(object sender, EventArgs e)
        {
            PermissionStatus status = await Permissions.CheckStatusAsync<Permissions.Microphone>();
            if (status is not PermissionStatus.Granted)
            {
                return;
            }
            StartRecording.IsEnabled = false;
            StopRecording.IsEnabled = true;

            await _audioRecorder.StartAsync();
        }
    }
}
```

Sadly I keep running in this error with no additional information I can go on to further debug the issue:
```
System.InvalidOperationException: 'Operation is not valid due to the current state of the object.'
```
![image](https://github.com/user-attachments/assets/c3e56b93-125c-4a06-a945-b65ae1eb4bdd)

And the stacktrace is as follow:
```
   at WinRT.ExceptionHelpers.<ThrowExceptionForHR>g__Throw|38_0(Int32 hr)
   at ABI.Windows.Storage.IApplicationDataStaticsMethods.get_Current(IObjectReference _obj)
   at Windows.Storage.ApplicationData.get_Current()
   at Plugin.Maui.Audio.AudioRecorder.<StartAsync>d__12.MoveNext()
   at MauiApp1.MainPage.<StartRecording_Clicked>d__4.MoveNext() in MainPage.xaml.cs:line 38
```

Any ideas what I could try or what could be the issue?

**Comment by nick4814** [comment]
I get this too for .NET 9 on windows 11

It fails on the line `await _audioRecorder.StartAsync();`

My code runs fine on android but not on windows

```
public async Task StartRecordingAsync()
{
    await _audioRecorder.StartAsync();
}
```

**Comment by nick4814** [comment]
I found the cause and tested a fix if any devs wanted to make a change. I am too lazy to do a pull request but leaving this here as a FYI. The issue happens in [AudioRecorder.windows.cs](https://github.com/jfversluis/Plugin.Maui.Audio/blob/ba47f6086fec6d7ee2a389abb502c148971ac3e2/src/Plugin.Maui.Audio/AudioRecorder/AudioRecorder.windows.cs#L28)

Fix:
![Image](https://github.com/user-attachments/assets/e76b5987-9b46-4912-9536-3edd3ed32bb7)

**Comment by bijington** [comment]
> I found the cause and tested a fix if any devs wanted to make a change. I am too lazy to do a pull request but leaving this here as a FYI. The issue happens in [AudioRecorder.windows.cs](https://github.com/jfversluis/Plugin.Maui.Audio/blob/ba47f6086fec6d7ee2a389abb502c148971ac3e2/src/Plugin.Maui.Audio/AudioRecorder/AudioRecorder.windows.cs#L28)
> 
> I am just using a private fork in the meantime with this fix: ![Image](https://github.com/user-attachments/assets/e76b5987-9b46-4912-9536-3edd3ed32bb7)

You've done all the hard work if you are using this on a private fork, why not just submit the Pull Request?

**Comment by nolios** [comment]
@bijington I have now submitted a [pull request](https://github.com/jfversluis/Plugin.Maui.Audio/pull/159) with this change in a new public fork

I made a public fork for this pull request as I had to add other logic needed for a private/personal project I am working on. There would be unnecessary code added to `Plugin.Audio.Maui` if I submitted a pull request on my private one

Additional things for you and other developers to take note of:
- The bug only showed up in my private project, the issue wasn't present in `Plugin.Maui.Audio.Sample` when I tested it
- I assume this bug is related to this [stack overflow post](https://stackoverflow.com/a/78396392), which could be due to the way some Maui Hybrid apps are setup (haven't confirmed this)

**Comment by jfversluis** [comment]
Yeah might have something to do with packages vs unpackaged and maybe Blazor Hybrid vs MAUI.

What type of project is the project you were running into this issue? Do you know/can you confirm if your fix works for both regular MAUI and Blazor Hybrid?

Thanks for sharing this!

**Comment by nolios** [comment]
@jfversluis The issue was ran into when I tested the library in a Blazor Hybrid project. Yeah I tested this fix in both MAUI/Blazor Hybrid projects and it works

From my testing, I also found that `FileSystem.AppDataDirectory` and `ApplicationData.Current.LocalFolder` evaluate to the same path too, so no change in the way this method works really. The following image demonstrates this and uses the `Plugin.Maui.Audio.Sample` MAUI project, where the bug isn't present and I could check both ways

![Image](https://github.com/user-attachments/assets/e1ff4e80-3df2-4bbf-8582-d88152592fc7)

No worries @jfversluis, happy to help! Great project 😃
